src/components/MovieCard/index.tsx

```diff
@@ -11,46 +11,48 @@ const MovieCard: React.FunctionComponent<IMovieCardProps> = ({ movie }) => {
   const { image, genre, name, year } = movie;
   const [isMouseOver, setIsMouseOver] = React.useState(false);
   const [isOptionsMenuOpen, setIsOptionsMenuOpen] = React.useState(false);
-  const [isEditMovieOpen, setIsEditMovieOpen] = React.useState(false);
-  const [isDeleteMovieOpen, setIsDeleteMovieOpen] = React.useState(false);
+
+  const setSelectedMovie = useSelectMovie();
+  const selectMovie = React.useCallback(
+    () => setSelectedMovie(movie),
+    [movie, setSelectedMovie]
+  );
+
+  const openDeleteModal = useDeleteModal();
+  const openEditModal = useEditModal();
 
   const optionsMenuItems = React.useMemo(
-    () =>
-      buildMenuItems(
-        setIsDeleteMovieOpen,
-        setIsEditMovieOpen,
-        setIsOptionsMenuOpen
-      ),
-    [setIsDeleteMovieOpen, setIsEditMovieOpen]
+    () => buildMenuItems(openDeleteModal, openEditModal, setIsOptionsMenuOpen),
+    [openDeleteModal, openEditModal]
   );
 
-  const showHoverEffect = React.useCallback(() => setIsMouseOver(true), []);
-  const hideHoverEffect = React.useCallback(() => setIsMouseOver(false), []);
+  const showHoverEffect = React.useCallback(
+    (): void => setIsMouseOver(true),
+    []
+  );
+  const hideHoverEffect = React.useCallback(
+    (): void => setIsMouseOver(false),
+    []
+  );
   const openOptionsMenu = React.useCallback(
-    () => setIsOptionsMenuOpen(true),
+    (): void => setIsOptionsMenuOpen(true),
     []
   );
   const closeOptionsMenu = React.useCallback(
-    () => setIsOptionsMenuOpen(false),
+    (): void => setIsOptionsMenuOpen(false),
     []
   );
```

Comment:
You might be able to get away with having one toggle function instead of this two functions.
Like: `... => setIsOptionsMenuOpen((prev) => !prev), []);`